### PR TITLE
Adding migration docs for 4.0.0

### PR DESCRIPTION
**Pull request by hsiaoer**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,335 @@ Change Log
...
+}
+
+To:
+
+maven {
+    url "https://fritz.mycloudrepo.io/public/repositories/android"
+}
+
+Changes
+* Adding support for model variants (fast, accurate, small)
```
Comment (ghop02) [inline CHANGELOG.md]: Can you add a couple details about these variants? Just explaining how they're different and the trade offs?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,335 @@ Change Log
...
+
+**Image Segmentation**
+
+* Renaming Classes ("Segment" -> "Segmentation"):
+    - FritzVisionSegmentPredictor -> FritzVisionSegmentationPredictor
+    - FritzVisionSegmentResult -> FritzVisionSegmentationResult
+    - FritzVisionSegmentPredictorOptions -> FritzVisionSegmentationPredictorOptions
+    - MaskType -> MaskClass
+
+* Model dependencies:
```
Comment (jamesonthecrow) [inline CHANGELOG.md]: We need to explain that dependencies have their own versioning system.

File: CHANGELOG.md (hunk 1 above)
Comment (jamesonthecrow) [inline CHANGELOG.md]: Also need to make a note about semantic model versioning.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,7 +5,341 @@ Change Log
...
+}
+
+To:
+
+maven {
+    url "https://fritz.mycloudrepo.io/public/repositories/android"
+}
+
+Changes
+* Adding support for model variants (fast, accurate, small).
```
Comment (jamesonthecrow) [inline CHANGELOG.md]: * Adding support for model variants (fast, accurate, small) so you can build the perfect experience for your users.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,7 +5,341 @@ Change Log
...
+maven {
+    url "https://fritz.mycloudrepo.io/public/repositories/android"
+}
+
+Changes
+* Adding support for model variants (fast, accurate, small). 
+     - Fast models are optimized for runtime performance with an accuracy tradeoff. This should be used in cases where model predictions need to happen quickly (e.g video processing, live preview, etc). This comes with a tradeoff in accuracy.
+     - Accurate models are optimized to display the best model prediction with a speed tradeoff. This should be used in cases where you're dealing with still images (i.e photo editing)
+     - Small models are optimized for model size at the cost of accuracy. This should be used in cases where developers are cautious of bloating their apps with models.
+* Models now have their own versioning system separate from the SDK and follow semantic versioning.
```
Comment (jamesonthecrow) [inline CHANGELOG.md]: Models now have their own versioning system separate from the SDK and follow semantic versioning. This enables Fritz to release new versions of models without changing any existing user experiences.

**Pull request merged**